articles/logging-with-winston/index.md

```diff
@@ -0,0 +1,547 @@
+
+This guide will explain logging within the context of Winston.
+
+### Why Bother with Logging
+- The only thing to know if something went wrong on the production app is to create logs. Log recreate and save that issue for you. If something is wrong or something is not working, the log will tell you.
+- To know how the system is behaving. Logging will generate information on how the system interacts with the users and what comes in and out of the system.
+- Keep track of your system activities. Logs go to the extent of showing when an instance happened and what triggered the log.
+
+Generally, the critical values of logging are:
+
+- Error tracking,
+- Debugging and
+- Application performance.
+
+### Why Choose Winston for Logging
+Winston is a famous logger with about weekly download. The following properties make Winston an overall universal logging middleware.
+
+- It is simple to use and configurable. Winston is simple to configure with your application.
+- Log [levels](https://github.com/winstonjs/winston#logging-levels) (priorities). Winston provides logging levels. They indicate the log priority; this gives you the ability to sort out critical logs from logs requiring less attention. For example: `{error: 0, warn: 1, info: 2, verbose: 3, debug: 4, silly: 5}`. In this case, an `error` log is given priority over a `verbose` log.
+- Logging [channels](https://github.com/winstonjs/winston#transports) (transports). A good logger has a different way of choosing your log output destination. With Winston, you can send and save logs in different ways, such as files, databases, emails, and console.
+- Log [formats](https://github.com/winstonjs/winston#formats). Winston provides you with several log formats. For example, when saving a log to a mongo database, the log format needs to be in JSON format. Winston solves that for you.
+- Log [profiling](https://github.com/winstonjs/winston#profiling). Winston helps you profile a block of code and measure the time the code takes to be successfully executed.
+
+### Using Winston
+
+To get started, make sure you initialize a Node.js project using `npm init`. We have a [guide](/engineering-education/beginner-guide-to-npm/) that will help you understand NPM in detail.
+
+Install the Winston package using the command `npm install winston`.
+
+Logging with Winston is simple, with just four steps as shown in the example below, and you have your a log recorded.
+
+- Add the Winston module with `require()` function.
+
+```js
+const winston = require('winston');
+```
+- Creating Winston configuration object. Winston requires at least one transport to create a log.
```

Comment:
I would explain here that transport is where the log is saved, file versus a database to be clearer to readers.